shared_model/backend/protobuf/queries/proto_ordering.hpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_SHARED_PROTO_MODEL_QUERY_ORDERING_HPP
+#define IROHA_SHARED_PROTO_MODEL_QUERY_ORDERING_HPP
+
+#include "interfaces/queries/ordering.hpp"
+#include "interfaces/common_objects/types.hpp"
+
+#include "queries.pb.h"
+
+namespace shared_model {
+  namespace proto {
+
+    /// Provides ordering data for pagination
+    class OrderingImpl final : public interface::Ordering {
+    public:
+      using BaseType = interface::Ordering;
```

Comment:
you can use ModelType
https://github.com/hyperledger/iroha/blob/10638156cfea555c989ea38e90989957ea1b67e1/shared_model/interfaces/base/model_primitive.hpp#L27

Reply:
How it will help the compiler to resolve Field and Direction types, which are presented in Ordering class?

Reply:
you can just use already existing `ModelType` instead of declaring new `BaseType` alias. e.g. `ModelType::Field`